### ISSUE TITLE
Unable to import settings from Firefox

### ISSUE THREAD

**Issue by SalahAdDin** [zen-browser/desktop#1023]
### What happened?

We are testing this browser as an alternative to Firefox; we want to import all data from Firefox, since it is our main browser, and Zen Browser is not Firefox, we thought it was possible. But it is not.

What about adding this feature?

Thank you

### Reproducible?

- [ ] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.0-a.31 (64-bit)

### Severity impact

Low

### What platform are you seeing the problem on?

Linux

**Comment by mithom** [comment]
duplicate of #221

**Comment by ben2talk** [comment]
Ok, just saw this and then noticed...

FIREFOX sync works.

This issue is to be expected - even with Firefox versions, profiles aren't always compatible - and corrupt profiles are best fixed by deleting and re-synchronising.

**Comment by SalahAdDin** [comment]
> Ok, just saw this and then noticed...
> 
> FIREFOX sync works, and manual import of bookmarks etc is okay too, so the impact is indeed just as low as, for example, with Firefox Nightly.
> 
> This issue is to be expected - even with Firefox versions, profiles aren't always compatible - and corrupt profiles are best fixed by deleting and re-synchronizing.

What about sessions?

**Comment by thibaultmol** [comment]
It would be nice to also have history import and also in general just profiles import from firefox

**Comment by julienallexandre** [comment]
Hello, why can't we import data from Firefox?

**Comment by 4cecoder** [comment]
> Hello, why can't we import data from Firefox?
> Firefox.to.Zen.mov

just want simple import from firefox nightly like @julienallexandre is trying to do

**Comment by kiiatto** [comment]
+1, running Firefox from pacman, Firefox is not listed in the Zen (appimage) import list

**Comment by MostHated** [comment]
I am using firefox-dev (aurora), and also have normal firefox installed but the only options in the zen import are  chrome, chromium, and vivaldi. While it's installed in a non-default location, I believe the profiles were auto created in the default location.

![](https://img.instance.id/hU93BaENV6Wg/direct)

![](https://img.instance.id/1zA9MehB4sAv/direct)

**Comment by lezetho** [comment]
+1, I would love this! Unfortunately until this is added I don't think I'll be moving to Zen... 😞

**Comment by vraravam** [comment]
The FF devs would not imagine a scenario where they will import settings, etc from one FF installation/version to another.

This being the case, I don't see this functionality being implemented in Zen - since this seems like a "core" functionality and Zen is built on top of FF.

**Comment by Swivelgames** [comment]
@vraravam That seems like a pretty big oversight for something that likely should be part of the core functionality in Zen. Right now, on its face, it appears that Zen only supports migrating from Chrome. So people who are still on Firefox don't get the same luxury?

Especially given that Zen is privacy focused, there are many of us that don't even use Firefox Sync. Granted, _downgrading_ profiles from a newer version of Firefox to an older one is likely problematic, but it's a bit disappointing that Zen likely wouldn't want FF users to migrate to Zen 😞

**Comment by vraravam** [comment]
@Swivelgames  - i don't think I made my observation clear (I am not associated with Zen except as another user just like you):

1. Zen is built on top of Firefox. **Without trivializing the efforts put by the Zen devs**, most of the features we experience are from FF, with some slight tweaks and enhancements. 
2. Now, in Firefox, the FF devs would not implement a feature to import settings, etc from FF itself - would they?

Based on the above, **my observation** was that such a feature would not exist. and i wouldn't hold my breath for the Zen devs to implement it either. If we get it, it would be good. But, if not, .....

As an aside, I was able to recreate most of my settings by following a few steps:
1. Shut down FF and Zen (at least run Zen once so that the profile folder is created)
2. open the profile folder in FF, and copy all files exactly as-is into the Zen profile folder.
3. with the above, most of the settings, etc came through anyways.
4. Some of the plugins that I use in FF have the capability to export their settings to a local file (for eg UBO). Export from FF, import into Zen.

Following the above steps, I was able to recreate exactly the same setup in Zen. After which I went into the settings in Zen and tweaked some more.

**Comment by Swivelgames** [comment]
@vraravam Absolutely. The solution that you outlined is what I had in mind as well. I imagine a more elegant solution that also works for newer versions of Firefox might not be necessary, if the Zen continues to keep strong parity with Firefox.

Understand, none of my comments are directed towards you personally.

The unfortunate reality is that #221 was closed for the very reasons outlined, and I think this is an unfortunate oversight. I'd love to see this implemented if possible.

**Comment by Swivelgames** [comment]
@SalahAdDin This might help.

It appears that there is an `--allow-downgrade` option. Not sure if this works or not, so YMMV.

![image](https://github.com/user-attachments/assets/8742f04c-a210-450e-a5ff-82b7a4307eca)

**Comment by neurokitti** [comment]
![image](https://github.com/user-attachments/assets/9c77883c-c778-46a2-8a8f-40b0e684dfb3)

still an issue

**Comment by mr-cheffy** [comment]
The reason this doesnt exist is because since we are based on firefox, why would firefox make an option to import data from... itself? 

Ill investigate more into this issue, but come on, "but it's a bit disappointing that Zen likely wouldn't want FF users to migrate to Zen 😞" where do you get this idea from? 😭

**Comment by mr-cheffy** [comment]
Im gonna save some links I saw from waterfox: 

https://github.com/BrowserWorks/Waterfox/pull/1553

**Comment by vraravam** [comment]
> The reason this doesnt exist is because since we are based on firefox, why would firefox make an option to import data from... itself?

This is exactly what I had responded a few weeks ago in reddit (unable to find my response now). Not sure, whether its the same person who created this ticket.

I agree - it would be absurd for FF folks to import from themselves!

**Comment by SalahAdDin** [comment]
> Im gonna save some links I saw from waterfox:
> 
> [BrowserWorks/Waterfox#1553](https://github.com/BrowserWorks/Waterfox/pull/1553)

They did it.

**Comment by thewbuk** [comment]
Any update on this? Is the only way to migrate from ff to zen to painstakingly copy all extensions and config? 

![Image](https://github.com/user-attachments/assets/09b66756-82ec-4961-856a-c98731d4cd39)

**Comment by Swivelgames** [comment]
> @mauro-balades
> ...but come on, "but it's a bit disappointing that Zen likely wouldn't want FF users to migrate to Zen 😞" where do you get this idea from? 😭

Here:

> @vraravam
> I don't see this functionality being implemented in Zen

**Comment by vraravam** [comment]
> Any update on this? Is the only way to migrate from ff to zen to painstakingly copy all extensions and config?

No, you can always follow these steps (for any FF-based browser)
1. Make sure your FF setup is completely to your liking
2. Quit/close FF (all windows)
3. Open and close Zen (this is so that the profile folder is created on your disk
4. Open the Profile folder from FF, and copy ALL files and folders into the profile folder that Zen created in the previous step
5. Start Zen

By following this, all FF preferences, plugins, themes, etc will be copied over (as of that particular point of time). Now, if you want any future FF settings, etc to be synced, you can use FF sync feature to do so. (just login into the same FF Sync account in both FF and Zen)

**Comment by vraravam** [comment]
Duplicate of #881 . I suggest we close this issue and continue conversations in a single place.

**Comment by Swivelgames** [comment]
@dosubot Not stale.

**Comment by SalahAdDin** [comment]
Yet interested in importing the settings from firefox

**Comment by mr-cheffy** [comment]
## Closing this issue as part of a repo cleanup

Due to technical-debt we had in the past, we are currently clearing out old, duplicate, stale, and non-actionable issues to improve issue tracking and make it easier to focus on fixing real bugs.

If this issue is still relevant, please open a new issue using our updated issue template, which will help us categorize and address it more efficiently.

Thanks for your understanding and for helping us improve the project! 🚀

This issue can be reopened at: https://github.com/zen-browser/desktop/issues/new?template=bug_report.yml&title=Unable%20to%20import%20settings%20from%20Firefox%0A

**Comment by SalahAdDin** [comment]
> ## Closing this issue as part of a repo cleanup
> 
> Due to technical-debt we had in the past, we are currently clearing out old, duplicate, stale, and non-actionable issues to improve issue tracking and make it easier to focus on fixing real bugs.
> 
> If this issue is still relevant, please open a new issue using our updated issue template, which will help us categorize and address it more efficiently.
> 
> Thanks for your understanding and for helping us improve the project! 🚀
> 
> This issue can be reopened at: https://github.com/zen-browser/desktop/issues/new?template=bug_report.yml&title=Unable%20to%20import%20settings%20from%20Firefox%0A

Wait a minute, this is not solved yet!

**Comment by mithom** [comment]
@SalahAdDin this ticket is a duplicate, keeping this one open is pointless. The other one has been closed for a long time (and linked in here!). any discussion around this topic should be done in the respective discussion as for now it isn't a bug/issue but controversial feature. To be discussed in the discussion around it ;)
https://github.com/zen-browser/desktop/discussions/881
